src/huggingface_hub/repocard.py

```diff
@@ -0,0 +1,38 @@
+import re
+from pathlib import Path
+from typing import Dict, Optional, Union
+
+import yaml
+
+
+# exact same regex as in the Hub server. Please keep in sync.
+REGEX_YAML_BLOCK = re.compile(r"---[\n\r]+([\S\s]*?)[\n\r]+---[\n\r]")
+
+
+def metadata_load(local_path: Union[str, Path]) -> Optional[Dict]:
+    content = Path(local_path).read_text()
+    match = REGEX_YAML_BLOCK.search(content)
+    if match:
+        yaml_block = match.group(1)
+        data = yaml.safe_load(yaml_block)
+        if isinstance(data, dict):
+            return data
+        else:
+            raise ValueError("repo card metadata block should be a dict")
+    else:
+        return None
```

Comment:
This should probably raise an error too, right? Running `metadata_load` on any random file will return nothing while I would expect it to fail if it doesn't find any metadata

Reply:
If that's the expected behavior then I would also put it in a test to ensure that it doesn't switch to raising an error in the future

Reply:
as discussed offline we have a lot of repo cards (model cards, etc) that do not have a yaml block, and i think the expected behavior here is to return None. If need be, we might modify this in a subsequent PR